src/preferences/metadatafilesettings.cpp

```diff
@@ -0,0 +1,189 @@
+#include "preferences/metadatafilesettings.h"
+
+#include <QAbstractItemView>
+#include <QApplication>
+#include <QFileDialog>
+#include <QLineEdit>
+#include <QMouseEvent>
+#include <QObject>
+#include <QTextCodec>
+#include <algorithm>
+
+#include "moc_metadatafilesettings.cpp"
+
+namespace {
+
+const ConfigKey kMetadataFileEnabled =
+        ConfigKey("[Livemetadata]", "MetadataFileEnabled");
+
+const ConfigKey kFileEncoding =
+        ConfigKey("[Livemetadata]", "FileEncoding");
+
+const ConfigKey kFileFormatString =
+        ConfigKey("[Livemetadata]", "FileFormatString");
+
+const ConfigKey kFilePath =
+        ConfigKey("[Livemetadata]", "CustomFormatString");
+
+const bool defaultFileMetadataEnabled = false;
+const QByteArray defaultEncoding = "UTF-8";
+const QString defaultFilePath = QDir::currentPath() + "/NowPlaying.txt";
+const QString defaultFileFormatString = "{{ track.artist }} - {{ track.title }}";
+
+} // namespace
+
+// static
+FileSettings MetadataFileSettings::s_latestSettings;
+// static
+const ConfigKey MetadataFileSettings::kFileSettingsChanged =
+        ConfigKey("[Livemetadata]", "FileSettingsChanged");
+
+MetadataFileSettings::MetadataFileSettings(UserSettingsPointer pSettings,
+        const FileWidgets& widgets,
+        QWidget* dialogWidget)
+        : m_pSettings(pSettings),
+          m_CPSettingsChanged(kFileSettingsChanged),
+          m_widgets(widgets),
+          m_pDialogWidget(dialogWidget),
+          m_fileEncodings{
+                  "UTF-8",
+                  "latin1",
```

Comment:
```suggestion
                  "Latin1",
```

Reply:
Qt uses "latin1" as alias for "ISO-8859-1". I have switched to the later.

Reply:
okay